hackathon_site/review/admin.py

```diff
@@ -1,4 +1,199 @@
 from django.contrib import admin
-from .models import Review
+from django.db.models import Count
+from django.utils.safestring import mark_safe
+from django.utils.translation import gettext_lazy as _
+
+from registration.models import Application
+from review.forms import ReviewForm, ApplicationReviewInlineFormset
+from review.models import Review, TeamReview
 
 admin.site.register(Review)
+
+
+class ApplicationInline(admin.TabularInline):
+    model = Application
+    autocomplete_fields = ("user",)
+    extra = 0
+    form = ReviewForm
+    formset = ApplicationReviewInlineFormset
+    can_delete = False
+    max_num = 0
+    exclude = (
+        "user",
+        "gender",
+        "ethnicity",
+        "phone_number",
+        "resume",
+        "rsvp",
+        "conduct_agree",
+        "data_agree",
+    )
+    readonly_fields = (
+        "get_user_full_name",
+        "study_level",
+        "school",
+        "graduation_year",
+        "q1",
+        "q2",
+        "q3",
+        "get_resume_link",
+        "birthday",
+        "get_reviewer_name",
+        "get_decision_sent_date",
+    )
+
+    def get_user_full_name(self, obj):
+        return f"{obj.user.first_name} {obj.user.last_name}"
+
+    get_user_full_name.short_description = "Full Name"
+
+    def get_resume_link(self, obj):
+        link = '<a href="{}" target="_blank">Open resume</a>'.format(obj.resume.url)
+        return mark_safe(link)
+
+    get_resume_link.short_description = "Resume"
+
+    def get_reviewer_name(self, obj):
+        if hasattr(obj, "review") and obj.review.reviewer:
+            return f"{obj.review.reviewer.first_name} {obj.review.reviewer.last_name}"
+
+        return ""
+
+    get_reviewer_name.short_description = "Reviewer name"
+
+    def get_decision_sent_date(self, obj):
+        if hasattr(obj, "review") and obj.review.decision_sent_date:
+            return obj.review.decision_sent_date
+
+        return ""
+
+    get_decision_sent_date.short_description = "Decision sent date"
+
+    def get_formset(self, request, obj=None, **kwargs):
+        """
+        Pass the request into the formset, so that we can then pass it down
+        into the form. Request is used to set the reviewer to the current user.
+        """
+        formset = super().get_formset(request, obj, **kwargs)
+        formset.request = request
+        return formset
+
+    def get_queryset(self, request):
+        return super().get_queryset(request).select_related("user", "review")
+
+    def has_change_permission(self, request, obj=None):
+        """
+        Let users with change_review permissions use this inline, rather than the default
+        change_application. Viewing still requires registration.view_application.
+        """
+        return request.user.has_perm("review.change_review")
+
+
+class TeamReviewedListFilter(admin.SimpleListFilter):
+    title = _("Team Reviewed")
```

Comment:
What's the point of using `gettext_lazy` for all of these?

Reply:
In case anyone wants to support translation for this project in the future. Other than that, no point that I know of (the docs just do it so I do too)